### PR TITLE
Add get diagnostic syntax to mysql dialect

### LINKED ISSUE #1148: Add Get Diagnostics syntax to mysql dialect
Add various keywords necessary to support `GET DIAGNOSTICS` functionality to the mysql dialect

### PR DESCRIPTION
**Pull request by silverbullettruck2001**
I also added 2 tests for code that I previously committed and was merged (`function_definer` and `procedure_definer`)

Resolves #1148

### CODE REVIEW

**Review by barrywhart — APPROVED** [review]
Looks good! A couple of observations, but nothing to hold up merging.

**Comment by barrywhart** [comment]
@silverbullettruck2001: Can you fix the `CHANGELOG.md` conflicts? Otherwise, this is good to merge. I'll be "away from keyboard" for a few days. Feel free to tag another maintainer to merge it.

**Review by barrywhart — CHANGES_REQUESTED** [review]
Small issue with the change log

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,14 +9,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- Support for select statement syntax specific to mysql dialect [#1175](https://github.com/sqlfluff/sqlfluff/issues/1175))
-- Support for the CALL statement for the mysql dialect [#1144](https://github.com/sqlfluff/sqlfluff/issues/1144))
+- Support for GET DIAGNOSTICS statement syntax specific to mysql dialect [#1148](https://github.com/sqlfluff/sqlfluff/issues/1148)
+- Support for select statement syntax specific to mysql dialect [#1175](https://github.com/sqlfluff/sqlfluff/issues/1175)
```
[the remark this answers is not in the document]
Comment (silverbullettruck2001) [reply CHANGELOG.md:14]: @barrywhart These duplicate lines have been removed.

**Review by barrywhart — APPROVED** [review]

**Pull request merged**